File: ParkingArea.cpp
#include "ParkingArea.h"
#include <iostream>
using namespace std;

ParkingArea::ParkingArea() : maxSlots(0), currentSlots(0), slots(nullptr) {}

ParkingArea::ParkingArea(const string& areaId, const string& zoneId, int maxSlots)
    : areaId(areaId), zoneId(zoneId), maxSlots(maxSlots), currentSlots(0) {
    slots = new ParkingSlot[maxSlots];
}

ParkingArea::~ParkingArea() {
    delete[] slots;
}

string ParkingArea::getAreaId() const {
    return areaId;
}

string ParkingArea::getZoneId() const {
    return zoneId;
}

int ParkingArea::getMaxSlots() const {
    return maxSlots;
}

int ParkingArea::getCurrentSlots() const {
    return currentSlots;
}

bool ParkingArea::addSlot(const string& slotId) {
    if (currentSlots >= maxSlots) {
        cout << "Error: Cannot add more slots. Area is full!" << endl;
        return false;
    }
    
    slots[currentSlots] = ParkingSlot(slotId, zoneId);
    currentSlots++;
    return true;
}

ParkingSlot* ParkingArea::findSlot(const string& slotId) {
    for (int i = 0; i < currentSlots; i++) {
        if (slots[i].getSlotId() == slotId) {
            return &slots[i];
        }
    }
    return nullptr;
}

ParkingSlot* ParkingArea::getFirstAvailableSlot() {
    for (int i = 0; i < currentSlots; i++) {
        if (slots[i].getAvailability()) {
            return &slots[i];
        }
    }
    return nullptr;
}

void ParkingArea::displayAllSlots() const {
    cout << "\n=== Area: " << areaId << " (Zone: " << zoneId << ") ===" << endl;
    cout << "Total Slots: " << currentSlots << "/" << maxSlots << endl;
    
    if (currentSlots == 0) {
        cout << "No slots created yet." << endl;
        return;
    }
    
    for (int i = 0; i < currentSlots; i++) {
        cout << i+1 << ". ";
        slots[i].displaySlotInfo();
    }
}

int ParkingArea::countAvailableSlots() const {
    int count = 0;
    for (int i = 0; i < currentSlots; i++) {
        if (slots[i].getAvailability()) {
            count++;
        }
    }
    return count;
}

File: python-flask-version/app/parking_area.py
from .parking_slot import ParkingSlot

class ParkingArea:
    def __init__(self, area_id=None, zone_id=None, max_slots=0):
        self.area_id = area_id
        self.zone_id = zone_id
        self.max_slots = max_slots
        self.current_slots = 0
        self.slots = []
    
    def get_area_id(self):
        return self.area_id
    
    def get_zone_id(self):
        return self.zone_id
    
    def get_max_slots(self):
        return self.max_slots
    
    def get_current_slots(self):
        return self.current_slots
    
    def add_slot(self, slot_id):
        """Add a slot to the area."""
        if self.current_slots >= self.max_slots:
            print("Error: Cannot add more slots. Area is full!")
            return False
        
        slot = ParkingSlot(slot_id, self.zone_id)
        self.slots.append(slot)
        self.current_slots += 1
        return True
    
    def find_slot(self, slot_id):
        """Find a slot by ID."""
        for slot in self.slots:
            if slot.get_slot_id() == slot_id:
                return slot
        return None
    
    def get_first_available_slot(self):
        """Get the first available slot in the area."""
        for slot in self.slots:
            if slot.get_availability():
                return slot
        return None
    
    def display_all_slots(self):
        """Display all slots in the area."""
        print(f"\n=== Area: {self.area_id} (Zone: {self.zone_id}) ===")
        print(f"Total Slots: {self.current_slots}/{self.max_slots}")
        
        if self.current_slots == 0:
            print("No slots created yet.")
            return
        
        for i, slot in enumerate(self.slots, 1):
            print(f"{i}. ", end="")
            slot.display_slot_info()
    
    def count_available_slots(self):
        """Count available slots in the area."""
        count = 0
        for slot in self.slots:
            if slot.get_availability():
                count += 1
        return count